crates/viewer/re_arrow_ui/src/datatype_ui.rs

```diff
@@ -0,0 +1,132 @@
+        DataType::List(field) => DataTypeUi {
+            type_name: "list".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::ListView(field) => DataTypeUi {
+            type_name: "list view".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::FixedSizeList(field, size) => DataTypeUi {
+            type_name: format!("fixed-size list ({size})"),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::LargeList(field) => DataTypeUi {
+            type_name: "large list".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::LargeListView(field) => DataTypeUi {
+            type_name: "large list view".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::Union(fields, mode) => {
+            let label = match mode {
+                arrow::datatypes::UnionMode::Sparse => "sparse union",
+                arrow::datatypes::UnionMode::Dense => "dense union",
+            };
+            DataTypeUi {
+                type_name: label.to_owned(),
+                content: Some(Box::new(move |ui| {
+                    for (_, field) in fields.iter() {
+                        data_type_field_ui(ui, field);
+                    }
+                })),
+            }
+        }
+        DataType::Dictionary(_k, _v) => DataTypeUi {
+            type_name: "dictionary".to_owned(),
+            content: None,
+        },
+        DataType::Map(a, _) => DataTypeUi {
+            type_name: "map".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, a);
+            })),
+        },
+        DataType::RunEndEncoded(a, b) => DataTypeUi {
+            type_name: "run-end encoded".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, a);
+                data_type_field_ui(ui, b);
+            })),
+        },
+        non_nested => {
+            let label = simple_data_type_string(non_nested)
+                .map(|s| s.to_owned())
+                .unwrap_or_else(|| non_nested.to_string());
+            DataTypeUi {
+                type_name: label,
+                content: None,
+            }
+        }
+    }
+}
+
+fn data_type_field_ui(ui: &mut egui::Ui, field: &arrow::datatypes::Field) {
+    ui.spacing_mut().item_spacing.y = 0.0;
+    let item = ui.list_item();
+
+    let data_type_ui = data_type_ui(field.data_type());
+
+    let property = PropertyContent::new(field.name())
+        .value_text(data_type_ui.type_name)
+        .show_only_when_collapsed(false);
```

Comment:
I think we should show the nullability of the field here too; see: 

* https://github.com/rerun-io/rerun/pull/10882